src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,343 @@
+    }
+
+
+    @Override
+    public void createDirectory(String name) throws DirectoryAlreadyExistException, FileManagerException {
+        File file = new File(name);
+        if (file.exists()) throw new DirectoryAlreadyExistException(file);
+        file.mkdir();
+        if (!file.exists()) throw new FileManagerException();
+    }
+
+    @Override
+    public void changeDirectory(String name) throws IllegalDirectoryException {
+        File file = new File(name);
+        if (!file.isDirectory()) {
+            throw new IllegalDirectoryException(file);
+        }
+        setCurrentDirectory(name);
+    }
+
+    @Override
+    public boolean delete(String name) throws FileNotFoundException {
+        File file = new File(name);
+        if (!file.exists()) {
+            throw new FileNotFoundException();
+        }
+        if (file.isDirectory()) {
+            if (file.list().length > 0) return false;
+        }
+        file.delete();
+        return true;
+
+    }
+
+
+    @Override
+    public boolean delete(String name, boolean force) throws FileNotFoundException {
+        File[] files = currentDirectory.listFiles();
+        File file = new File(name);
+        if (!force) {
+            delete(name);
+        } else {
+            for (File f : files) {
+                f.delete();
+            }
+        }
+        file.delete();
+        return true;
+    }
+
+    @Override
+    public void copy(String source, String dest, CopyMode copyMode) throws
+            FileNotFoundException, IllegalArgumentException, FileAlreadyExistException, FileCopyException {
+        switch (copyMode) {
+            case SIMPLE:
+                try {
+                    File sourceFile = new File(source);
+                    File destFile = new File(dest);
+                    if (!(sourceFile.exists())) throw new FileNotFoundException();
+                    if (sourceFile.isDirectory()) throw new IllegalArgumentException();
+                    if (destFile.exists()) throw new FileAlreadyExistException(destFile);
+                    else destFile.createNewFile();
+                    FileInputStream fin = new FileInputStream(source);
+                    FileOutputStream fout = new FileOutputStream(dest);
+                    int i;
+                    do {
+                        i = fin.read();
+                        if (i != -1)
+                            fout.write(i);
+                    }
+                    while (i != -1);
+                    fin.close();
+                    fout.close();
+                } catch (IOException n) {
+                }
+                break;
+            case BUFFERED:
+                copy(source, dest);
+                break;
+        }
+    }
+
+    @Override
+    public void copy(String source, String dest) throws
+            FileNotFoundException, IllegalArgumentException, FileAlreadyExistException, FileCopyException {
+        try {
+            File sourceFile = new File(source);
```

Comment:
Move buffered implementation into other `copy` method and call it from this one using:
```
copy(source,dest, CopyMode.BUFFERED)
```